### PR TITLE
feat: Add Partitioned Parameter to SetCookie

### PR DESCRIPTION
**Pull request by Haswf**
#### What type of PR is this?

feat

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) Translate the PR title into Chinese.

feat: SetCookie 增加 Partitioned 参数

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en: 
As discuessed, this PR adds `partitioned` parameter to (*RequestContext).SetCookie, simplifying the use of partitioned cookies.

zh(optional):

#### (Optional) The PR that updates user documentation:

Yes, see https://github.com/cloudwego/cloudwego.github.io/pull/929 for related doc changes.

### CODE REVIEW

File: pkg/app/context.go [hunk 1]
```diff
@@ -1218,15 +1218,17 @@ func (ctx *RequestContext) Cookie(key string) []byte {
...
-//	3. ctx.SetCookie("", "hertz", 10, "/", "localhost",protocol.CookieSameSiteLaxMode, false, false)
+//	3. ctx.SetCookie("", "hertz", 10, "/", "localhost",protocol.CookieSameSiteLaxMode, false, false, false)
 //	add response header --->  Set-Cookie: hertz; max-age=10; domain=localhost; path=/; SameSite=Lax;
-//	4. ctx.SetCookie("user", "", 10, "/", "localhost",protocol.CookieSameSiteLaxMode, false, false)
+//	4. ctx.SetCookie("user", "", 10, "/", "localhost",protocol.CookieSameSiteLaxMode, false, false, false)
 //	add response header --->  Set-Cookie: user=; max-age=10; domain=localhost; path=/; SameSite=Lax;
-func (ctx *RequestContext) SetCookie(name, value string, maxAge int, path, domain string, sameSite protocol.CookieSameSite, secure, httpOnly bool) {
+//	5. ctx.SetCookie("user", "name", 10, "/", "localhost",protocol.CookieSameSiteNoneMode, true, true, true) add
+//	response header Set-Cookie: user=name; max-age=10; domain=localhost; path=/; HttpOnly; secure; SameSite=None; Partitioned
+func (ctx *RequestContext) SetCookie(name, value string, maxAge int, path, domain string, sameSite protocol.CookieSameSite, secure, httpOnly, partitioned bool) {
```
Comment (welkeyever) [inline pkg/app/context.go:1231]: 这个地方可能是一个 break，考虑新增一个 SetPartitionedCookie方法？包一层 SetCookie 可能是一个温和的支持手段？

Comment (Haswf) [reply pkg/app/context.go:1231]: 那把实现放在 setCookie 里， SetPartitionedCookie和SetCookie 传不同的参数？

Comment (welkeyever) [reply pkg/app/context.go:1231]: 对，这样或许平滑一些？

Comment (Haswf) [reply pkg/app/context.go]: 好的，已经按照这样设计了

**Review by welkeyever — APPROVED** [review]
LGTM, thx~

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**